src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int figuresCount = 3;
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        FigureType randomFigureType = FigureType.values()[new Random().nextInt(figuresCount)];
+
+        switch (randomFigureType) {
+            case SQUARE:
+                figure = new Square(getRandomIn100to0Range(),
+                        colorSupplier.getRandomColor());
+                break;
+            case RIGHT_TRIANGLE:
+                figure = new RightTriangle(getRandomIn100to0Range(),
+                        getRandomIn100to0Range(),
+                        colorSupplier.getRandomColor());
+                break;
+            case RECTANGLE:
+                figure = new Rectangle(getRandomIn100to0Range(),
+                        getRandomIn100to0Range(),
+                        colorSupplier.getRandomColor());
+                break;
+            case ISOSCELES_TRAPEZOID:
+                figure = new IsoscelesTrapezoid(getRandomIn100to0Range(),
+                        getRandomIn100to0Range(),
+                        getRandomIn100to0Range(),
+                        colorSupplier.getRandomColor());
+                break;
+            default:
+                figure = new Circle(getRandomIn100to0Range(),
+                        colorSupplier.getRandomColor());
+                break;
+        }
+
+        return figure;
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, Color.WHITE.name());
+    }
+
+    private int getRandomIn100to0Range() {
```

Comment:
this method generates a random integer between 1 (inclusive) and 100 (exclusive). please, change method name accordingly or fix the code to avoid misunderstandings